src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,134 @@
+                String password = request.queryParams("password");
+                User user = usersList.get(name);
+                if(password.equals(user.password)) {
+                    Session session = request.session();
+                    session.attribute("user", name);
+                    response.redirect("/messages");
+                }
+            return null;
+            }
+                ));
+
+        Spark.post("/create-user",
+                ((request, response) -> {
+                    String name = request.queryParams("user");
+                    String password = request.queryParams("password");
+                    User user = new User(name, password, new ArrayList<Message>());
+                    usersList.put(name, user);
+                    Session session = request.session();
+                    session.attribute("user", name);
+                    response.redirect("/messages");
+                    return "";
+
+                }
+                ));
+
+        Spark.post("/create-message",
+                ((request, response) -> {
+                    //get session out of request
+                    Session session = request.session();
+                    //get 'user' attribute from session that we stored earlier (in either /login or /create-user)
+                    String name = session.attribute("user");
+                    //get user object from the userslist hashmap using 'name' as the key
+                    User user = usersList.get(name);
+
+
+                    //get the text of our message from the request parameters
+                    String message = request.queryParams("message");
+                    //use the text of our message to create a new message object
+                    Message m = new Message(message);
+                    //store our message in the user object's messagelist
+                    user.messageList.add(m);
+
+
+                    //send user back to /messages
+                    response.redirect("/messages");
+                    return "";
+
+                })
+
+        );
+
+        Spark.post("/logout",
+                ((request, response) -> {
+                   Session session = request.session();
+                    session.invalidate();
+                    response.redirect("/");
+                    return "";
+
+
+                }));
+
+        Spark.post("/delete", ((request, response) -> {
+
+            Session session = request.session();
+            String name = session.attribute("user");
+            User user = usersList.get(name);
+            String m = request.queryParams("delete");
+            int i = Integer.parseInt(m) - 1;   //i converted a string to an integer here so i could delete the message
+                    user.messageList.remove(i);
+            //send user back to /messages
+            response.redirect("/messages");
+            return "";
+
+
+        })
```

Comment:
Everything looks great so far Eric.  You just need to add the logic for update message.  That means a form similar to the delete form.  It will need an additional text field to bring in the text the user wants to update to.  Then add the corresponding updateMessage post route in Main to do the actual updating of the message.  Add that logic and resubmit.